### DISCUSSION TITLE
How to templatize Resource Discovery configuration file ?

### DISCUSSION THREAD

**Discussion by testskills** [tomkerkhove/promitor#1905] [Q&A]
Hi,

I wanted to know how can we templatize Resource Discovery configuration file? Like we have resource group names vary based on the environment's and also Tags value can also change which are used in the Criteria. Is it possible to create placeholders in the config file and have values inputted from another yml or properties file.

resourceDiscoveryGroups:
- name: sql-database-landscape
  type: SqlDatabase
  criteria:
    include:
     resourceGroups:
     - dev-resourceGroup1  **--> This can change to prod-resourceGroup1**
     - dev-resourceGroup2  **--> This can change to prod-resourceGroup2**
     - dev-resourceGroup3  **--> This can change to prod-resourceGroup3**

**Comment by testskills** [comment]
Following up on this one if @tomkerkhove or anyone has any thoughts.

**Reply by tomkerkhove** [reply]
Sorry for the delay but I'm on holiday.

You can use regular expressions when using tags - https://docs.promitor.io/configuration/v2.x/resource-discovery#criteria

Hope that helps? Otherwise, a feature request is the best approach as we don't support regex for resource groups or subscriptions - Sorry :/

**Reply by testskills** [reply]
Thank you for your response. I will create feature request. It's good to have regex at least for resource groups.
